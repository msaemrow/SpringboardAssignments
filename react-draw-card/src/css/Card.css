.Card{
    margin: 0 5px;
}

.Card-name{
    margin-bottom: 0;
    font-weight: 400;
    color: white;
}

.Card-image{
    height: 190px;
    width: auto;
}