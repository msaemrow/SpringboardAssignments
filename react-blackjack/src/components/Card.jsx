import React from "react";
import "../css/Card.css"

function Card(props){
    return(
        <img 
        className="Card" 
        src={`https://deckofcardsapi.com/static/img/${props.cardId}.png`} 
        alt="Playing Card"
        />
    )
}

export default Card;