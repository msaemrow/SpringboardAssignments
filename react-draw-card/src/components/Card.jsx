import React from "react";
import "../css/Card.css"

function Card({ value, image }){
    return(
        <div className="Card">
            <p className="Card-name">{value}</p>
            <img className="Card-image" src={image} alt={value} />
        </div>
    )
}

export default Card;