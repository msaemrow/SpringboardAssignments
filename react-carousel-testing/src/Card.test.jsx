import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";
import { it } from "vitest";

it("loads Card without crashing", () => {
    render(<Card
      caption={TEST_IMAGES[0].caption}
      src={TEST_IMAGES[0].caption}
      currNum={0}
      totalNum={0}
    />);  
  });


  it('matches snapshot', () => {
    const { asFragment } = render(
      <Card
      caption={TEST_IMAGES[0].caption}
      src={TEST_IMAGES[0].caption}
      currNum={0}
      totalNum={0}
    />);
    expect(asFragment()).toMatchSnapshot();
  })
