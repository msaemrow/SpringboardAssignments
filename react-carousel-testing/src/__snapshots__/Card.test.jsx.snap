// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="testing image 1"
    />
    <small
      class="Card-small"
    >
      Image 0 of 0.
    </small>
  </div>
</DocumentFragment>
`;
